libraries/commerce/favorites/subscriptions/index.js

```diff
@@ -15,23 +19,38 @@ export default function favorites(subscribe) {
     return;
   }
 
+  /** App start / favorites route enter */
   subscribe(shouldFetchFavorites$, ({ dispatch }) => {
     dispatch(fetchFavorites());
   });
 
+  /** User login / logout */
   subscribe(shouldFetchFreshFavorites$, ({ dispatch }) => {
     dispatch(fetchFavorites(true));
   });
 
-  /*
-   * Request after 5 seconds since last sync request to make sure
-   * backend did actually save it.
+  /**
+   * Request after N seconds since last sync request to make sure
+   * backend did actually save it
    */
-  let fetchThrottle;
-  subscribe(favoritesSyncIdle$, ({ dispatch }) => {
-    clearTimeout(fetchThrottle);
-    fetchThrottle = setTimeout(() => {
-      dispatch(fetchFavorites(true));
-    }, FETCH_FAVORITES_THROTTLE);
+  const refreshFavorites$ = favoritesSyncIdle$.debounceTime(FETCH_FAVORITES_THROTTLE);
+  subscribe(refreshFavorites$, ({ dispatch }) => {
+    dispatch(fetchFavorites(true));
+  });
+
+  const errorQuotaFavorites$ = favoritesError$.filter(({ action }) => (
+    action.error && action.error.code === ELIMIT
+  ));
+  subscribe(errorQuotaFavorites$, ({ dispatch, action }) => {
+    dispatch(showModal({
+      confirm: null,
+      dismiss: 'modal.ok',
+      title: 'modal.title_error',
+      message: 'favorites.error_limit',
+    })).then(() => {
+      if (action.productId) {
+        dispatch(requestRemoveFavorites(action.productId, true));
```

Comment:
It looks like this will write `isFetching = true` to the state (in the reducer) and no one clears it afterwards.

Reply:
Fixed